Zend/zend_vm_def.h

```diff
@@ -3574,6 +3607,21 @@ ZEND_VM_HOT_OBJ_HANDLER(112, ZEND_INIT_METHOD_CALL, CONST|TMPVAR|UNUSED|THIS|CV,
 			}
 			HANDLE_EXCEPTION();
 		}
+		if (UNEXPECTED(
+			((fbc->common.fn_flags & ZEND_ACC_MUTATING) != 0)
+			!= ((opline->extended_value & ZEND_INIT_METHOD_CALL_MUTATING) != 0)
```

Comment:
I had to double read this. It's not two clauses, but a simple (in)equality one. Maybe there is a more obvious way of writing this?

Reply:
Might be a bit of a brain twister ^^ It's an abbreviated form of:

```c
if (UNEXPECTED(
	((fbc->common.fn_flags & ZEND_ACC_MUTATING) && !(opline->extended_value & ZEND_INIT_METHOD_CALL_MUTATING))
	|| (!(fbc->common.fn_flags & ZEND_ACC_MUTATING) && (opline->extended_value & ZEND_INIT_METHOD_CALL_MUTATING))
) {}

Reply:
Would you prefer:

```c
if (UNEXPECTED(
	(bool)(fbc->common.fn_flags & ZEND_ACC_MUTATING)
	!= (bool)(opline->extended_value & ZEND_INIT_METHOD_CALL_MUTATING)
) {}
```

Reply:
Seems more cleaner